### PR TITLE
Nrfx none run wdt error cases on nrf54h

### PR DESCRIPTION
**Pull request by nordic-segl**
Add WDT instances 010, 011, 131 and 132 that exist in nRF54H20.
Enable execution of tests/drivers/watchdog/wdt_error_cases on nRF54H20.

### CODE REVIEW

File: dts/common/nordic/nrf54h20.dtsi [hunk 1]
```diff
@@ -168,6 +168,20 @@
 				interrupts = <64 NRF_DEFAULT_IRQ_PRIORITY>,
 					     <65 NRF_DEFAULT_IRQ_PRIORITY>;
 			};
+
+			wdt010: watchdog@14000 {
```
Comment (57300) [inline dts/common/nordic/nrf54h20.dtsi:172]: Please use unique labels in this file, such as:
```suggestion
			cpuapp_wdt010: watchdog@14000 {
```
Then, you can add localized labels to the following files in `dts/arm/nordic`:

```
/* nrf54h20_cpuapp.dtsi */

wdt010: &cpuapp_wdt010 {};
wdt011: &cpuapp_wdt011 {};

/* nrf54h20_cpurad.dtsi */

wdt010: &cpurad_wdt010 {};
wdt011: &cpurad_wdt011 {};
```

**Review by 57300 — CHANGES_REQUESTED** [review]

File: dts/common/nordic/nrf54h20.dtsi (hunk 1 above)
Comment (nordic-segl) [reply dts/common/nordic/nrf54h20.dtsi]: I applied changes. Check if this is what You meant.

**Review by 57300 — DISMISSED** [review]

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml [hunk 2]
```diff
@@ -14,3 +14,5 @@ flash: 512
 supported:
   - gpio
   - pwm
+  - spi
```
Comment (anangl) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:17]: This change is not related to watchdogs.

File: dts/common/nordic/nrf54h20.dtsi [hunk 3]
```diff
@@ -168,6 +168,20 @@
 				interrupts = <64 NRF_DEFAULT_IRQ_PRIORITY>,
 					     <65 NRF_DEFAULT_IRQ_PRIORITY>;
 			};
+
+			cpuapp_wdt010: watchdog@14000 {
+				compatible = "nordic,nrf-wdt";
+				reg = <0x14000 0x1000>;
```
Comment (anangl) [inline dts/common/nordic/nrf54h20.dtsi:174]: Add also corresponding entries in `validate_base_addresses.c`.

File: dts/common/nordic/nrf54h20.dtsi [hunk 4]
```diff
@@ -293,6 +307,20 @@
 				status = "disabled";
 				interrupts = <61 NRF_DEFAULT_IRQ_PRIORITY>;
 			};
+
+			cpurad_wdt010: watchdog@13000 {
```
Comment (anangl) [inline dts/common/nordic/nrf54h20.dtsi:311]: Please keep the order by node address.

File: tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpurad.overlay [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&uart135 {
+    pinctrl-0 = < &uart136_default >;
+    pinctrl-1 = < &uart136_sleep >;
+};
```
Comment (anangl) [inline tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpurad.overlay:10]: Why?

File: tests/drivers/watchdog/wdt_error_cases/src/main.c [hunk 6]
```diff
@@ -51,6 +51,15 @@
 #define MAX_INSTALLABLE_TIMEOUTS (8)
 #define WDT_WINDOW_MAX_ALLOWED   (0x07CFFFFFU)
 #define DEFAULT_OPTIONS          (WDT_OPT_PAUSE_IN_SLEEP | WDT_OPT_PAUSE_HALTED_BY_DBG)
+#elif defined(CONFIG_SOC_NRF54H20)
```
Comment (anangl) [inline tests/drivers/watchdog/wdt_error_cases/src/main.c:54]: Why not use `#if defined(CONFIG_SOC_NRF54L15) || defined(CONFIG_SOC_NRF54H20)`? The configuration seems to be the same.

File: tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&wdt010 {
+   status = "okay";
+};
+
+&wdt011 {
+   status = "disabled";
+};
```
Comment (anangl) [inline tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:13]: Disabling nodes disabled by default is redundant.

File: tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&wdt010 {
+   status = "okay";
+};
```
Comment (anangl) [inline tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:9]: nit:
```suggestion
&wdt010 {
	status = "okay";
};
```

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml (hunk 2 above)
Comment (nordic-segl) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:17]: I moved it to a separate commit. This change is needed to run following configuration in Twister:
https://github.com/zephyrproject-rtos/zephyr/pull/71677

File: dts/common/nordic/nrf54h20.dtsi (hunk 3 above)
Comment (nordic-segl) [reply dts/common/nordic/nrf54h20.dtsi:174]: This is a black magic for me.
I found following defines in HAL
modules/hal/nordic:
nrfx/mdk/nrf54h20_radiocore.h:    #define NRF_WDT010                            NRF_RADIOCORE_WDT010
nrfx/mdk/nrf54h20_radiocore.h:    #define NRF_WDT010                            NRF_RADIOCORE_WDT010
nrfx/mdk/nrf54h20_application.h:    #define NRF_WDT010                            NRF_APPLICATION_WDT010
nrfx/mdk/nrf54h20_application.h:    #define NRF_WDT010                            NRF_APPLICATION_WDT010

File: dts/common/nordic/nrf54h20.dtsi (hunk 4 above)
Comment (nordic-segl) [reply dts/common/nordic/nrf54h20.dtsi:311]: Done

File: tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay (hunk 8 above)
Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:9]: I changed 4xSPACE to a single TAB.

File: tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay (hunk 7 above)
Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:13]: Idea behind this is to have a list of WDT instances that can be easily "switched" ON/OFF.

File: tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpurad.overlay (hunk 5 above)
Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/boards/nrf54h20dk_nrf54h20_cpurad.overlay:10]: I'm not 100% sure this should be here. I add this because:
- I run Twister on App and Radio using the same UART port (on PC side); So it was easier for me to have both cores printing on the same GPIOs. It's enought to run twister twice, each run on other cpu with different UART port. 
- I used secdom FW with STM traces; it prints logs on GPIOs defined in pincontrol for uart135.

I will remove this to have all tests on Radio core printing on the same UART.

File: tests/drivers/watchdog/wdt_error_cases/src/main.c (hunk 6 above)
Comment (nordic-segl) [reply tests/drivers/watchdog/wdt_error_cases/src/main.c:54]: I thought that nrf54l15 supports WDT_FLAG_RESET_SOC
while nrf54h20 supports WDT_FLAG_RESET_CPU_CORE.
Therefore I started with separate configurations.
Test results show that I was wrong.

I will apply change as suggested.

File: dts/common/nordic/nrf54h20.dtsi (hunk 3 above)
Comment (anangl) [reply dts/common/nordic/nrf54h20.dtsi:174]: I think we should use `CHECK_DT_REG(cpuapp_wdt010, NRF_APPLICATION_WDT010);` and so on.

Comment (nordic-segl) [reply dts/common/nordic/nrf54h20.dtsi:174]: Done. Please confirm it's all what's needed.

**Review by anangl — APPROVED** [review]

File: dts/common/nordic/nrf54h20.dtsi (hunk 3 above)
Comment (anangl) [reply dts/common/nordic/nrf54h20.dtsi:174]: LGTM now, thanks.

File: dts/common/nordic/nrf54h20.dtsi [hunk 9]
```diff
@@ -168,6 +168,20 @@
...
+				reg = <0x14000 0x1000>;
+				status = "disabled";
+				interrupts = <20 NRF_DEFAULT_IRQ_PRIORITY>;
+			};
+
+			cpuapp_wdt011: watchdog@15000 {
+				compatible = "nordic,nrf-wdt";
+				reg = <0x15000 0x1000>;
+				status = "disabled";
+				interrupts = <21 NRF_DEFAULT_IRQ_PRIORITY>;
```
Comment (nordic-segl) [inline dts/common/nordic/nrf54h20.dtsi:183]: _The peripheral ID is the interrupt number (IRQ number)
The peripheral ID is given by bits [22:12] of a peripheral address_
Shall I switch IRQ number to 0x15 or can it stay in decimal?

Comment (anangl) [reply dts/common/nordic/nrf54h20.dtsi:183]: The way it is now is consistent with other nodes, so let it stay like this.

Comment (nordic-segl) [reply dts/common/nordic/nrf54h20.dtsi:183]: Remaining nodes have IRQ given in decimal form.

**Review by nordic-krch — APPROVED** [review]
